### PR TITLE
feat: VSCode WebView Initialization with development and build scripts

### LINKED ISSUE #60: [feature] VSCode WebView Initialization with development and build scripts

### LINKED ISSUE #62: Use ts scripts to run tests, and don't have to compile them into js (to be discussed)

### PR DESCRIPTION
**Pull request by SoonIter**
closes #60 closes #62 

## Details

- VSCode WebView Initialization in `src/view.ts`
- build script -> tsup
	- why bundle and not tsc? 
		- to replace `process.env.NODE_ENV`
		- to run in the browser in the future [working-with-extensions/bundling-extension](https://code.visualstudio.com/api/working-with-extensions/bundling-extension)

- dev script -> webpack + react-refresh  
	- inspired by https://github.com/tjx666/vscode-webview-webpack-hmr-example

-  type-safe communication between client and webview -> unport
	- implemented with https://github.com/web-infra-dev/unport
- some ui packages
	- react-use
	- @chakra-ui/react
	- react-icons

## Review Step

click "Run And Debug" in vscode and "Launch  Client", enjoy the hmr in the webview

image

### CODE REVIEW

**Comment by HerringtonDarkholme** [comment]
Hi thanks for the explanation. This pull request and explanation is very informative. Thanks.

> build script -> tsup
> dev script -> webpack + react-refresh

I think migrating them to Rspack later will save two separate deps and runs faster. 

But we can leave it as is now and migrating later.

File: scripts/dev/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const { resolve } = require('path')
+const { HotModuleReplacementPlugin } = require('webpack')
+const ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin')
+
+const devServerClientOptions = {
+  hot: true,
+  // !: 指定构造 WebSocket 的协议是 ws
```
Comment (HerringtonDarkholme) [inline scripts/dev/webpack.config.js:7]: nit: remove

**Review by HerringtonDarkholme — APPROVED** [review]
Looks nice. component library choice is nice. 

Would you like to resolve conflicts and push update?

**Comment by SoonIter** [comment]
> Looks nice. component library choice is nice.
> 
> Would you like to resolve conflicts before it can be merged?

Of course, dealing with lock file conflicts is one of my strengths in a sense.

To ensure that the ci is green when this pr is merged, I used tsx to run the test script in advance.Therefore, one more commit was pushed.

**Pull request merged**